agent/agent.go

```diff
@@ -97,21 +97,27 @@ func (a *Agent) loop(ctx context.Context) error {
 	timer := time.NewTimer(0)
 	defer timer.Stop()
 	var tc taskContext
+	var exit bool
 	for {
 		select {
 		case <-ctx.Done():
 			grip.Info("agent loop canceled")
 			return nil
 		case <-timer.C:
-			nextTask, err := a.comm.GetNextTask(ctx)
+			nextTask, err := a.comm.GetNextTask(ctx, &apimodels.GetNextTaskDetails{tc.taskGroup})
 			if err != nil {
 				return errors.Wrap(err, "error getting next task")
 			}
 			if nextTask.TaskId != "" {
 				if nextTask.TaskSecret == "" {
 					return errors.New("task response missing secret")
 				}
-				tc = a.prepareNextTask(ctx, nextTask, &tc)
+				tc, exit = a.prepareNextTask(ctx, nextTask, &tc)
+				if exit {
+					// Query for next task, this time with an empty task group,
+					// to get a ShouldExit from the API, and set NeedsNewAgent.
+					continue
```

Comment:
use loop levels. this continue just continues the select statement not the for loop, which might be the same, but could be confusing in the future.

Reply:
fixed